src/ol/interaction/Modify.js

```diff
@@ -439,27 +444,24 @@ class Modify extends PointerInteraction {
     }
     const map = this.getMap();
     if (map && map.isRendered() && this.getActive()) {
-      this.handlePointerAtPixel_(this.lastPixel_, map);
+      this.handlePointerAtPixel_(map.getCoordinateFromPixel(this.lastPixel_));
     }
     feature.addEventListener(EventType.CHANGE, this.boundHandleFeatureChange_);
   }
 
   /**
    * @param {import("../MapBrowserEvent.js").default} evt Map browser event.
-   * @param {Array<Array<SegmentData>>} segments The segments subject to modification.
+   * @param {Array<SegmentData>} segments The segments subject to modification.
    * @private
    */
   willModifyFeatures_(evt, segments) {
     if (!this.featuresBeingModified_) {
       this.featuresBeingModified_ = new Collection();
       const features = this.featuresBeingModified_.getArray();
       for (let i = 0, ii = segments.length; i < ii; ++i) {
-        const segment = segments[i];
-        for (let s = 0, ss = segment.length; s < ss; ++s) {
-          const feature = segment[s].feature;
-          if (feature && !features.includes(feature)) {
-            this.featuresBeingModified_.push(feature);
-          }
+        const feature = segments[i].feature;
+        if (feature && !features.includes(feature)) {
+          this.featuresBeingModified_.push(feature);
```

Comment:
This change is not directly related - in trying to understand (again) what the Modify interaction does, I added the `DragSegment` typedef above, and using that type for the `dragSegments_` member revealed a problem with the code. While the nested loop did work, it made the code unnecessarily hard to read, and required nested arrays to be passed as the `willModifyFeatures_`'s `segments` argument.